cpp/src/arrow/flight/sql/example/sqlite_statement_batch_reader.cc

```diff
@@ -0,0 +1,177 @@
+  case TYPE_CLASS##Type::type_id: {                                               \
+    int bytes = sqlite3_column_bytes(STMT, COLUMN);                               \
+    const unsigned char* string = sqlite3_column_text(STMT, COLUMN);              \
+    if (string == nullptr) {                                                      \
+      ARROW_RETURN_NOT_OK(                                                        \
+          (reinterpret_cast<TYPE_CLASS##Builder&>(builder)).AppendNull());        \
+      break;                                                                      \
+    }                                                                             \
+    ARROW_RETURN_NOT_OK(                                                          \
+        (reinterpret_cast<TYPE_CLASS##Builder&>(builder)).Append(string, bytes)); \
+    break;                                                                        \
+  }
+
+#define BINARY_BUILDER_CASE(TYPE_CLASS, STMT, COLUMN)                                  \
+  case TYPE_CLASS##Type::type_id: {                                                    \
+    int bytes = sqlite3_column_bytes(STMT, COLUMN);                                    \
+    const void* blob = sqlite3_column_blob(STMT, COLUMN);                              \
+    if (blob == nullptr) {                                                             \
+      ARROW_RETURN_NOT_OK(                                                             \
+          (reinterpret_cast<TYPE_CLASS##Builder&>(builder)).AppendNull());             \
+      break;                                                                           \
+    }                                                                                  \
+    ARROW_RETURN_NOT_OK(                                                               \
+        (reinterpret_cast<TYPE_CLASS##Builder&>(builder)).Append((char*)blob, bytes)); \
+    break;                                                                             \
+  }
+
+#define INT_BUILDER_CASE(TYPE_CLASS, STMT, COLUMN)                        \
+  case TYPE_CLASS##Type::type_id: {                                       \
+    sqlite3_int64 value = sqlite3_column_int64(STMT, COLUMN);             \
+    ARROW_RETURN_NOT_OK(                                                  \
+        (reinterpret_cast<TYPE_CLASS##Builder&>(builder)).Append(value)); \
+    break;                                                                \
+  }
+
+#define FLOAT_BUILDER_CASE(TYPE_CLASS, STMT, COLUMN)                      \
+  case TYPE_CLASS##Type::type_id: {                                       \
+    double value = sqlite3_column_double(STMT, COLUMN);                   \
+    ARROW_RETURN_NOT_OK(                                                  \
+        (reinterpret_cast<TYPE_CLASS##Builder&>(builder)).Append(value)); \
+    break;                                                                \
+  }
+
+#define MAX_BATCH_SIZE 1024
```

Comment:
Can you use a `constexpr` instead? e.g. `static constexpr int kMaxBatchSize = 1024`.

(also, put it insinde the namespaces below)